nvm.sh

```diff
@@ -335,25 +335,30 @@ nvm_find_nvmrc() {
   fi
 }
 
-# Obtain nvm version from rc file
+# Obtain the nvm version from the .nvmrc file
 nvm_rc_version() {
+  local NVM_USE_SILENT
+  NVM_USE_SILENT=false
+  if [ "${1-}" = '--silent' ]; then
+    NVM_USE_SILENT=true
+  fi
   export NVM_RC_VERSION=''
   local NVMRC_PATH
   NVMRC_PATH="$(nvm_find_nvmrc)"
   if [ ! -e "${NVMRC_PATH}" ]; then
-    if [ "${NVM_SILENT:-0}" -ne 1 ]; then
+    if [ "${NVM_USE_SILENT}" != true ]; then
       nvm_err "No .nvmrc file found"
     fi
     return 1
   fi
-  NVM_RC_VERSION="$(command head -n 1 "${NVMRC_PATH}" | command tr -d '\r')" || command printf ''
+  NVM_RC_VERSION="$(command cat "${NVMRC_PATH}" | command tr -d '\r' | sed -e 's/\s*#.*$//' | sed -e '/^\s*$/d' | command tr -d '\n' | command tr -d '\t' | command head -n 1 | sed -e 's/[[:space:]]//g')" || command printf ''
```

Comment:
# 
Recently, I had a `.nvmrc` file inside one of my projects. As you know, most projects have only one, and I wanted to add in some comments. These line changes above, are the main part of this pull request. Instead of running `head` on the file, I replaced it for `cat`. Next, the next two `sed` commands remove comments. I also removed newlines and tabs inside. After that, I ran `head -n 1` on everything to get only the first line. In the case of the following scenario:
```bash
node # This is the nvm version
```
There is going to be a space thereafter where it says `node`. Next, I also removed spaces for that scenario. That is a quick read of what is happening in the main part of the pull request.
#

Reply:
right, but why did you want to add comments? The file is named "nvm rc", who needs the extra hint that it's for nvm?

Reply:
@ljharb those are good puritan points, and perhaps you're right, but I think in our case on a large project, where people don't even necessarily know what nvm is let alone an `nvmrc` it can be helpful to explain e.g. "# we're fixing to 16.2.0 because of this bug <link to bug>" and or simply re-document the default behaviours "this version will be used by default when you run nvm install. It's an agreed upon, tested & supported node version for the project." Coordinating communications on projects with people with a wide range of backgrounds/seniority can mean that documenting "what" is helpful.

Reply:
Isn’t that what git is for?

Reply:
I agree, but I don't think expecting my interns to read a git lens comment is as easy as having them read the file contents. 

Should we just put all our efforts into teaching interns the correct methodologies and stop perpetuating incompetence in our industry? Probably, yes. 

On the other hand, if the world is covered in unnecessary thorns, is it easier to wear shoes than of rid the world of thorns? Sometimes the pragmatic option is the correct option. 

Anyway - you make good points, and I can appreciate why you'd say no. JSON doesn't have comments either.

Reply:
@ljharb I can maybe put good example - in comment there could be link to issue that require to keep node in specific version due to some complications something like - for now we have this comment on CI workflows with node version - i would like to move it into `nvmrc` without loosing information why it is locked to particular version.
```
v16.10.0 # https://github.com/facebook/jest/issues/11956
```
But as keeping `nvm` simple is also good valid point.